src/components/visualizations/chart/highcharts/plugins/dataLabelsColors.ts

```diff
@@ -37,15 +39,34 @@ function setLabelsColor(chart: any) {
     });
 }
 
+function setHeatMapDataLabelsColor(chart: any) {
+    const dataClasses = get(chart, 'colorAxis.0.dataClasses', []);
+    const points = flatMap(getVisibleSeries(chart), (series: any) => series.points)
```

Comment:
same construction as in setBarLabelsColor, move into separate function eg. getVisiblePointsWithLabel and use on both places